src/lock/digest.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{fmt, str::FromStr};
+
+use serde::{de::Visitor, Deserialize, Serialize};
+use thiserror::Error;
+
+/// Supported types of digest algorithms.
+// Do not reorder variants; the ordering is significant, see #38 and #106.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum DigestAlgorithm {
+    /// SHA-2 with 256 bits
+    #[serde(rename = "sha256")]
+    SHA256,
+}
+
+/// Error parsing a [`DigestAlgorithm`].
+#[derive(Error, Debug)]
+pub enum DigestAlgorithmError {
+    /// Represents a ring digest algorithm that isn't supported by Buffrs
+    #[error("unsupported digest algorithm: {0}")]
+    UnsupportedAlgorithm(String),
+}
+
+impl FromStr for DigestAlgorithm {
+    type Err = DigestAlgorithmError;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        match serde_typename::from_str(input) {
+            Ok(value) => Ok(value),
+            _other => Err(DigestAlgorithmError::UnsupportedAlgorithm(input.into())),
+        }
+    }
+}
+
+#[test]
```

Comment:
Sorry I'm late to review, but I don't think this is good practice @xfbs. First time I've seen tests outside of a module gated by `#[cfg(test)]` and after [reviewing the docs](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-module-and-cfgtest) I'm pretty sure there's a good reason for that.